security/libaes_siv/Portfile

```diff
@@ -0,0 +1,48 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+# NOTE: Doesn't work with cmake 1.1 PortGroup
+
+PortSystem          1.0
+PortGroup           github 1.0
+PortGroup           cmake 1.0
+
+github.setup        dfoxfranke libaes_siv 1.0.0 v
+revision            0
+categories          security devel
+maintainers         {fwright.net:fw @fhgwright} openmaintainer
+
+description         A C implementation of AES-SIV
+
+long_description    This is an RFC5297-compliant C implementation of AES-SIV, \
+                    written by Daniel Franke on behalf of Akamai Technologies \
+                    and published under the Apache License (v2.0). It uses \
+                    OpenSSL for the underlying AES and CMAC implementations \
+                    and follows a similar interface style.
+
+license             apache-2
+platforms           darwin
+
+github.tarball_from releases
+master_sites        https://github.com/${github.author}/${name}/archive
+distname            v${github.version}
+worksrcdir          ${name}-${github.version}
```

Comment:
These variables should be set up already by the github portgroup if you use it correctly.

Reply:
Nope.  I started with just "github.tarball_from releases", and then added the others as needed to make it actually work.  There seem to be some discrepancies in the naming between the upstream code and the expectations of the github portgroup.  Specifically:

1. For some reason, the tarball comes from the 'archive' directory, rather than the 'release' directory.

1. The tarball name does not include the project name, so the distname needs to be changed from the default in order to get the correct tarball name.

1. After overriding the distname to get the correct tarball name, it's then necessary to override the resulting worksrcdir to make it match the paths _within_ the archive, essentially changing it _back to_ what it would have been without the distname override.

I didn't try to figure out whether there's a way to change the tarball name without impacting worksrcdir, which could save one of the three overrides.

Reply:
> Nope. I started with just "github.tarball_from releases", and then added the others as needed to make it actually work

Can't the distname can be handled by setting the prefix you already set on the github.setup line?

Reply:
As you already noted- the packages doesn't come from the 'release' directory so therefore I think you shouldn't use the ```github.tarball_from releases```. If you remove that line then you can also remove the ```master_sites```, ```distname```, and ```worksrcdir``` as @pmetzger suggested. You'll need to update the checksums.

Reply:
Yes, the somewhat "cookbooky" documentation on the github PortGroup doesn't make it clear that the sole function of `github.tarball_from` is to provide one of four canned setups for `github.master_sites`, and hence is useless in this case.  Just setting `github.master_sites` to reflect the 'archive' setup is sufficient.  The earlier trouble may have been due to setting `master_sites` instead of `github.master_sites`.

Omitting the (new) `github.master_sites` line still successfully fetches *a* tarball, but one with a slightly different length (and of course different checksums) than the "official release" tarball, which is clearly less desirable behavior.  It's always best to stick to official release tarballs as much as possible, which is why `github.tarball_from releases` is recommended in the documentation, even though it doesn't actually work in this case.

Reply:
see PR #2587 that would address downloading from 'archives'

Reply:
I don't really want to wait for a 5-month-old PR (not to mention a 5-year-old ticket) to be resolved before publishing this port. :-) Based on what I've empirically determined to work here, the proposed enhancement to `github.tarball_from` isn't quite right as it stands, anyway.  In fact, what I currently have is very similar to the `releases` case except for a different constant "middle part" of `github.master_sites`, and the result doesn't have the issues Ryan raised in #2587.  There's at least one port (archivers/quazip) which does exactly the same thing as I'm doing here; there are probably others that could do so if they knew better.

Implementing that change (properly) would be a convenience, but not really a simplification, since it would just replace one line of a Portfile with another.  The main benefit would be in people's not having to figure out the recipe.